src/plugins/layout/shadowdom.ts

```diff
@@ -36,7 +36,7 @@ export class ShadowDom {
     let shadowNode = this.doc.createElement("div") as IShadowDomNode;
     shadowNode.id = "" + index;
     shadowNode.node = node;
-    shadowNode.ignore = shouldIgnoreNode(node, this);
+    shadowNode.ignore = shouldIgnoreNode(node) || (parent && parent.ignore && parent.node !== document);
```

Comment:
> || (parent && parent.ignore && parent.node !== document); [](start = 46, length = 58)

Move this condition to function.

Reply:
I moved it to a separate function. I want to avoid state provider knowing about shadowdom to simplify things here and check for parent needs to lookup shadowdom, which I am trying to avoid.